### PR TITLE
Provide some context on why our connection fails

### PR DESCRIPTION
**Pull request by thomaseizinger**
Instead of just sending a `ReadFail` message, we send along an error
that gives us more context on why reading from the socket failed.

Found as part of production hat because the existing log message was fairly poor and didn't help me in diagnosing.

### CODE REVIEW

File: daemon/src/maker_inc_connections.rs [hunk 1]
```diff
@@ -417,26 +418,40 @@ impl Actor {
...
 
         let mut tasks = Tasks::default();
-        tasks.add({
-            let this = this.clone();
-
-            async move {
-                while let Ok(Some(msg)) = read.try_next().await {
-                    let res = this
-                        .send(maker_cfd::FromTaker {
+        tasks.add_fallible(
```
Comment (luckysori) [inline daemon/src/maker_inc_connections.rs:421]: ❓ Is the advantage of using `add_fallible` here that we are more explicit about the error case? Or is there a functional difference?

**Review by luckysori — APPROVED** [review]

File: daemon/src/maker_inc_connections.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs:421]: It allows me to use `?` inside the future and handle it all in one spot!

Comment (luckysori) [reply daemon/src/maker_inc_connections.rs:421]: Ah, of course! Well worth it.

**Pull request merged**